Source/WebCore/PAL/pal/system/glib/SleepDisablerGLib.cpp

```diff
@@ -91,8 +92,13 @@ void SleepDisablerGLib::acquireInhibitor()
         g_variant_builder_init(&builder, G_VARIANT_TYPE_VARDICT);
         g_variant_builder_add(&builder, "{sv}", "reason", g_variant_new_string(m_reason.utf8().data()));
         parameters = g_variant_new("(su@a{sv})", "" /* no window */, 8 /* idle */, g_variant_builder_end(&builder));
+    } else if (const gchar* prgname = g_get_prgname()) {
+        parameters = g_variant_new("(ss)", prgname, m_reason.utf8().data());
+    } else if (const auto executablePath = FileSystem::currentExecutablePath(); !executablePath.isNull()) {
+        GUniquePtr<char> executableName(g_path_get_basename(executablePath.data()));
+        parameters = g_variant_new("(ss)", executableName.get(), m_reason.utf8().data());
     } else
-        parameters = g_variant_new("(ss)", g_get_prgname(), m_reason.utf8().data());
+        parameters = g_variant_new("(ss)", "", m_reason.utf8().data());
```

Comment:
Looking at the [API docs](https://specifications.freedesktop.org/idle-inhibit/latest/#org.freedesktop.ScreenSaver.Inhibit) there's no clear evidence that an empty string should be allowed, so I wouldn't even attempt this. Even if it works in GNOME, it might not work in another desktop. Just return early here.